### ISSUE TITLE
Comment cS64_C

### ISSUE THREAD

**Issue by demurgos** [ShyykoSerhiy/skyweb#31]
Hi,
[cS64_C](https://github.com/ShyykoSerhiy/skyweb/blob/master/src/utils.ts#L102) is a very cryptic function. The variable names are too short to mean anything (except that there's a hash and stuff with 64 bits).
I understand that hashing functions usually look like this, but could you at least add a comment about how you found this function ? Is it with some Reverse Engineering ?
I googled `cS64_C` and variants and wasn't able to find a standard algorithm called this way. What doew qw mean ?

Could you clarify this ?

**Comment by demurgos** [comment]
Still in the same file: does `getMac256Hash` relates to [HMAC](https://en.wikipedia.org/wiki/Hash-based_message_authentication_code) ?
If so, why is it called "256" when you return 128 bits (4 `uint32`) ?

**Comment by demurgos** [comment]
Dude !
`cS64_C` does not even work !
`bigInt` does not modify the current value but returns a new instance ! [All these lines do nothing !!!](https://github.com/ShyykoSerhiy/skyweb/blob/master/src/utils.ts#L127-139)

``` typescript
// this...
qwDatum.multiply(CS64_e);
qwDatum.mod(mod);
// is this:
qwDatum * CS64_e;
qwDatum % mod;
// and NOT:
qwDatum *= CS64_e;
qwDatum = qwDatum  % mod;
```

**Comment by ShyykoSerhiy** [comment]
Hi, I've grubbed this code from Microsoft Research Cryptography Library.

**Comment by ShyykoSerhiy** [comment]
And it seems that your comment about bigInt is correct. Have no idea how it works then :)

**Comment by demurgos** [comment]
Which file did you used ? I also downloaded the MSR-crypto lib.

I wrote a simple decompiler for SkypeWeb and got the [sha256Auth.js file](https://github.com/Demurgos/skype-web-reversed/blob/fe3931c4f091af06f6b2c2e8c14608aebf87448b/skype/latest/decompiled/fullExperience/rjs%24%24msr-crypto/lib/sha256Auth.js#L48) because I was not satisfied.
It's minimified, but still legible so I refactored the `utils.ts` to comment it and make it more straight-forward. I'm currently writing a test-suite to ensure that it works fine, I'll send a PR this evening or tomorrow.

**Comment by demurgos** [comment]
Ok, I started to fix issues on my fork, but there is just too much to do so I just created a "hard" fork to develop [my own lib](https://github.com/demurgos/skype-http) on top of yours.
I added linting, unit-tests, CI, fixed the hmac computation, wrote type definitions for dependencies, progressively refactor it to actually **use** Typescript (static type checks, interfaces, imports, ES6, etc.), abstract IO, clean URI building, etc.
It's not finished yet and I don't exclude an eventual merge to this package again, but for the moment this repo does not seem active enough.
